Mailroom_2 for review

scotchwsplenda opened the pull request.

natasha-aleksandrova commented on students/scotchwsplenda/Lesson_4/MailRoom_2.py:
```diff
@@ -0,0 +1,106 @@
+        elif response  =="4":
+            mass_mail()
+        else:
+            print("Not a valid option!")
```
per instructions, you should be using a switch dict here

natasha-aleksandrova commented on students/scotchwsplenda/Lesson_4/MailRoom_2.py:
```diff
@@ -0,0 +1,106 @@
+        response = input(prompt)  # continuously collect user selection
+        # now redirect to feature functions based on the user selection
+        if response == "1":
+            Send_Note()
```
should be lower case

natasha-aleksandrova commented on students/scotchwsplenda/Lesson_4/MailRoom_2.py:
```diff
@@ -0,0 +1,106 @@
+        "\n"+"Indicate your choice:  ")
+        if respondy=="1":
+            for key, value in donators.items():
+                print(key)
```
if you're only interested in keys, this is better:
```
for donor_name in donators:
    print(donor_name)
```

natasha-aleksandrova commented on students/scotchwsplenda/Lesson_4/MailRoom_2.py:
```diff
@@ -0,0 +1,106 @@
+                new_don = input("That's an unknown donor, input donation amount: ")
+                new_don=float(new_don)
+                donators.update({donor_inp:[new_don]})
+                send_thanks(donor_inp,new_don)
```
there is some duplicate code in lines 62-71 - how could you condense it?

natasha-aleksandrova commented on students/scotchwsplenda/Lesson_4/MailRoom_2.py:
```diff
@@ -0,0 +1,106 @@
+# Option 2: create report
+def data_metrics():
+    report = []
+    for name, donations in donators:
```
this is an invalid syntax and crashes:
```
$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
Welcome to the MAILROOM main menu
Please choose from below options:
1 - Initiate Groveling
2 - Data Metrics 3000
3 - Quit
4 - Send thank you notes to all donors
$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
Indicate your choice:  2
Traceback (most recent call last):
  File "student.py", line 106, in <module>
    main()
  File "student.py", line 34, in main
    data_metrics()
  File "student.py", line 78, in data_metrics
    for name, donations in donators:
ValueError: too many values to unpack (expected 2)
```

scotchwsplenda commented:
Don't know if this email reply is necessary or if the pull request
automatically notify's you .This is for a new better grade please.
fixed the reporting (needed to add '.items' after changing data collection
object to a dictionary. Added a switch_dict instead of elif for main menu.
Cleaned up other random stuff.

https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/674

On Tue, Jan 14, 2020 at 8:44 PM Natasha <notifications@github.com> wrote:

> Merged #670
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/670>
> into master.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/UWPCE-PythonCert-ClassRepos/SP_Online_PY210/pull/670?email_source=notifications&email_token=AM2M4SY6KNSWKUMOTKCAF2TQ52IERA5CNFSM4KGKRVOKYY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOV7GYOFI#event-2949482261>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AM2M4S3IHJ25CXWAKWBG64LQ52IERANCNFSM4KGKRVOA>
> .
>


-- 
*Oliver B. Lockwood*

natasha-aleksandrova commented:
Got it 👍 thank you for letting us know
